### DISCUSSION TITLE
How to create and use an utility script?

### DISCUSSION THREAD

**Discussion by rubisara38** [freeplane/freeplane#459] [Help] [answered]
I followed [the guide here](https://www.freeplane.org/wiki/index.php/Your_own_utility_script_library) to create several utility methods inside an utility script.

But when I tried to call one of these methods in my main script, I got this error:

![fp_error](https://user-images.githubusercontent.com/58072905/174431697-93318294-c83b-418e-ab85-ede54be3ec8a.png)

(In the error picture above, SearchGUI is my main script, Utils is the ulitity script.)

Strange thing is: if I just defined a constant in the utility script and used this constant in the main script, then it was ok, no error.

Please help me. Thanks.

**Comment by macmarrum** [comment]
Interesting.
It would help me to see the actual scripts that fail.
Could you please zip them and attach them here? It can be a cut down version, if you'd like to avoid sharing the whole thing.

**Comment by rubisara38** [comment, marked as answer]
Hi @macmarrum thank you for your answer. I finally solved my problem.

There were several reasons for the difficulties I have faced:

(I listed them here so that this might help someone else if they ran into the same problem in the future):

1. My Utils script is not auto-recompiled after I make changes to it (and call it again). Instead, it is only recompiled when I make changes to the caller script (ie. the script that uses the Utils script).

2. I was confused by IntelliJ marking my variable declarations and usages as errors, although in reality they are correct:

![groovy_intellij_error](https://user-images.githubusercontent.com/58072905/174476624-ecd3400f-095e-44e0-afcb-bc9799a04cac.png)

3. The error message has given by Freeplane (which I attached on my first post here) is not very informative.
